src/deepsparse/benchmark/torchscript_engine.py

```diff
@@ -0,0 +1,325 @@
+        """
+        return None
+
+    @property
+    def input_names(self) -> List[str]:
+        """
+        :return: The ordered names of the inputs.
+        """
+        return [node_arg.name for node_arg in self._eng_net.get_inputs()]
+
+    @property
+    def input_shapes(self) -> List[Tuple]:
+        """
+        :return: The ordered shapes of the inputs.
+        """
+        return [tuple(node_arg.shape) for node_arg in self._eng_net.get_inputs()]
+
+    @property
+    def output_names(self) -> List[str]:
+        """
+        :return: The ordered names of the outputs.
+        """
+        return [node_arg.name for node_arg in self._eng_net.get_outputs()]
+
+    @property
+    def output_shapes(self) -> List[Tuple]:
+        """
+        :return: The ordered shapes of the outputs.
+        """
+        return [tuple(node_arg.shape) for node_arg in self._eng_net.get_outputs()]
+
+    def run(
+        self,
+        inp: List[numpy.ndarray],
+        val_inp: bool = True,
+    ) -> List[numpy.ndarray]:
+        """
+        Run given inputs through the model for inference.
+        Returns the result as a list of numpy arrays corresponding to
+        the outputs of the model as defined in the ONNX graph.
+
+        Note 1: the input dimensions must match what is defined in the torch.nn.Module.
+        To avoid extra time in memory shuffles, the best use case
+        is to format both the Module and the input into channels first format;
+        ex: [batch, height, width, channels] => [batch, channels, height, width]
+
+        Note 2: the input type for the numpy arrays must match
+        what is defined in torch.nn.Module
+        Generally float32 is most common,
+        but int8 and int16 are used for certain layer and input types
+        such as with quantized models.
+
+        Note 3: the numpy arrays must be contiguous in memory,
+        use numpy.ascontiguousarray(array) to fix if not.
+
+        | Example:
+        |     engine = TorchScriptEngine("path/to/.pt", batch_size=1, num_cores=None)
+        |     inp = [numpy.random.rand(1, 3, 224, 224).astype(numpy.float32)]
+        |     out = engine.run(inp)
+        |     assert isinstance(out, List)
+
+        :param inp: The list of inputs to pass to the engine for inference.
+            The expected order is the inputs order as defined in the ONNX graph.
+        :param val_inp: Validate the input to the model to ensure numpy array inputs
+            are setup correctly for inference.
+        :return: The list of outputs from the model after executing over the inputs
+        """
+
+        torch_inputs = [torch.from_numpy(input).to(self.device) for input in inp]
+
+        tensors = self._model(*torch_inputs)
+
+        if isinstance(tensors, torch.Tensor):
+            tensors = [tensors]
+        if self.device == "cuda":
+            tensors = [torch.Tensor.cpu(tensor) for tensor in tensors]
+        return [tensor.detach().numpy() for tensor in tensors]
```

Comment:
should always be safe to call `cpu` on a tensor, even if not in `cuda` mode
```suggestion
        return [tensor.cpu().detach().numpy() for tensor in tensors]
```